server/src/domain/sql_backend_handler.rs

```diff
@@ -87,7 +93,7 @@ pub mod tests {
             .unwrap();
     }
 
-    pub async fn insert_user_no_password(handler: &SqlBackendHandler, name: &str) {
+    pub async fn insert_user_no_password<A: BackendHandler>(handler: &A, name: &str) {
```

Comment:
I'm not going to review this commit, it feels like it'll change a lot after our conversation on the trait/handler wrapping.